include/CoreDisTools/coredistools.h

```diff
@@ -0,0 +1,36 @@
+//===--------- coredistools.h - Dissassembly tools for CoreClr
+//-------------===//
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// \brief Core Disassembly Tools API for CoreClr AOT/JIT
+///
+//===----------------------------------------------------------------------===//
+
+#if !defined(LLILC_TOOLS_COREDISTOOLS)
+#define LLILC_TOOLS_COREDISTOOLS
+
+#if !defined(DllIface)
+#define DllIface extern "C" __declspec(dllimport)
```

Comment:
I would expect this header to be compatible with C unless there is some convention discouraging that that I'm not aware of (possible).  Constructs like `extern "C"` need a `defined(__cplusplus)` guard.  Also, is `__declspec(dllimport)` relevant/appropriate on non-Win?